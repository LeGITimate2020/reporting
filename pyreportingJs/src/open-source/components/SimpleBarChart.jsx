import React from 'react';
import PropTypes from 'prop-types';
import { PlotlyComponent } from '../index';

/**
 * Pure React component to create a Plotly bar chart.
 * <SimpleBarChart xValues={['thing1', 'thing2', 'thing3']} yValues={[12, 13.9, 1]}  />
 */
const SimpleBarChart = ({ xValues, yValues, layout }) => {
  const _layout = {
    margin: {
      t: 10,
    },
    height: 300,
    showlegend: true,
    legend: { orientation: 'h' },
    ...layout,
  };
  const config = {
    displaylogo: false,
    modeBarButtonsToRemove: ['sendDataToCloud'],
  };
  const data = [
    {
      type: 'bar',
      x: xValues,
      y: yValues,
    },
  ];

  return (
    <PlotlyComponent
      data={data}
      layout={_layout}
      config={config}
    />
  );
};

SimpleBarChart.propTypes = {
  xValues: PropTypes.array.isRequired,
  yValues: PropTypes.arrayOf(PropTypes.number).isRequired,

  /** Plotly layout options */
  layout: PropTypes.object,
};

SimpleBarChart.defaultProps = {
  layout: {},
};

export default SimpleBarChart;
