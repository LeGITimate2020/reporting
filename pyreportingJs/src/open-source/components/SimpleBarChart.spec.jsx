import React from 'react';
import { shallow } from 'enzyme';
import SimpleBarChart from './SimpleBarChart';
import { PlotlyComponent } from '../index';

describe('SimpleBarChart', () => {
  const props = {
    xValues: ['bar1', 'bar2', 'bar3'],
    yValues: [0, 1, 2],
  };

  it('renders without crashing!', () => {
    shallow(<SimpleBarChart {...props} />);
  });

  it('Contains one plotly component with correct props', () => {
    const wrapper = shallow(<SimpleBarChart {...props} />);
    expect(wrapper.find(PlotlyComponent).length).toBe(1);
  });

  it('Matches snapshot', () => {
    const wrapper = shallow(<SimpleBarChart {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
