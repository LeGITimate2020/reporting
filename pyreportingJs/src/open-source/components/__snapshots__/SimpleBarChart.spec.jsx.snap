// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleBarChart Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SimpleBarChart
    layout={Object {}}
    xValues={
      Array [
        "bar1",
        "bar2",
        "bar3",
      ]
    }
    yValues={
      Array [
        0,
        1,
        2,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoscaleVisibleYAxis": true,
      "config": Object {
        "displaylogo": false,
        "modeBarButtonsToRemove": Array [
          "sendDataToCloud",
        ],
      },
      "data": Array [
        Object {
          "type": "bar",
          "x": Array [
            "bar1",
            "bar2",
            "bar3",
          ],
          "y": Array [
            0,
            1,
            2,
          ],
        },
      ],
      "layout": Object {
        "height": 300,
        "legend": Object {
          "orientation": "h",
        },
        "margin": Object {
          "t": 10,
        },
        "showlegend": true,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoscaleVisibleYAxis": true,
        "config": Object {
          "displaylogo": false,
          "modeBarButtonsToRemove": Array [
            "sendDataToCloud",
          ],
        },
        "data": Array [
          Object {
            "type": "bar",
            "x": Array [
              "bar1",
              "bar2",
              "bar3",
            ],
            "y": Array [
              0,
              1,
              2,
            ],
          },
        ],
        "layout": Object {
          "height": 300,
          "legend": Object {
            "orientation": "h",
          },
          "margin": Object {
            "t": 10,
          },
          "showlegend": true,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
